src/tests/blueprint/t_blueprint_mesh_transform.cpp

```diff
@@ -452,37 +456,90 @@ TEST(conduit_blueprint_mesh_transform, polygonal_transforms)
+            const conduit::int64_array poly_subdata = poly_subconn_array.as_int64_array();
+            
+            conduit::Node poly_size;
+            poly_size = topo_poly["elements/sizes"];
+
+            // BHAN - Error when trying to convert empty poly_subsize,
+            // set to element/sizes for polygonal (unused)
+            conduit::Node poly_subsize;
+            if (is_topo_3d)
+            {
+                poly_subsize = topo_poly["subelements/sizes"];
+            }
+            else 
+            {
+                poly_subsize = topo_poly["elements/sizes"];
+            }
+
+            conduit::Node poly_size_array;
+            poly_size.to_int64_array(poly_size_array);
+            const conduit::int64_array poly_size_data = poly_size_array.as_int64_array();
+
+            conduit::Node poly_subsize_array;
+            poly_subsize.to_int64_array(poly_subsize_array);
+            const conduit::int64_array poly_subsize_data = poly_subsize_array.as_int64_array();
+
             for(index_t ep = 0, et = 0; ep < poly_len;
                 ep += poly_stride, et += topo_indices)
             {
-                EXPECT_EQ(poly_data[ep], is_topo_3d ? topo_faces : topo_findices);
+                EXPECT_EQ(poly_size_data[ep / poly_stride],
+                          is_topo_3d ? topo_faces : topo_findices);
 
-                for(index_t efo = ep + is_topo_3d; efo < ep + poly_stride;
-                    efo += 1 + topo_findices)
+                for(index_t efo = ep; efo < ep + poly_stride;
+                    efo += is_topo_3d ? 1 : topo_findices)
                 {
-                    EXPECT_EQ(poly_data[efo], topo_findices);
+                    EXPECT_EQ(is_topo_3d ? poly_subsize_data[efo / poly_stride] :
+                                           poly_size_data[efo / poly_stride],
+                                           topo_findices);
 
                     const std::set<index_t> topo_index_set(
                         &topo_data[et],
                         &topo_data[et + topo_indices]);
-                    const std::set<index_t> poly_index_set(
-                        &poly_data[efo + 1],
-                        &poly_data[efo + 1 + topo_findices]);
+
+                    std::set<index_t> poly_index_set;
+                    if (is_topo_3d)
+                    {
+                        std::set<index_t> polyhedral_index_set(
+                        &poly_subdata[poly_data[efo] * topo_findices],
+                        &poly_subdata[poly_data[efo] * topo_findices + topo_findices]);
+                        poly_index_set = polyhedral_index_set;
+                    }
+                    else
+                    {
+                        std::set<index_t> polygonal_index_set(
+                        &poly_data[efo],
+                        &poly_data[efo + topo_findices]);
+                        poly_index_set = polygonal_index_set;
```

Comment:
Please indent the contents of this conditional and the previous conditional to make the code easier to follow.